test/vfd_swmr_bigset_writer.c

```diff
@@ -875,64 +1381,99 @@ verify_dset_attribute(hid_t ds, unsigned int which, unsigned int step)
-        errx(EXIT_FAILURE, "H5Aread failed");
+    return true;
 
-    if (H5Aclose(aid) < 0)
-        errx(EXIT_FAILURE, "H5Aclose failed");
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Aclose(aid);
+    }
+    H5E_END_TRY;
 
-    if (read_step != step)
-        errx(EXIT_FAILURE, "expected %u read %u", step, read_step);
+    return false;
 }
 
-static void
-verify_extensible_dset(state_t *s, unsigned int which, mat_t *mat, unsigned *stepp)
+static bool
+verify_extensible_dset(state_t *s, unsigned int which, mat_t *mat, unsigned finished_step, unsigned last_step)
 {
-    hid_t        ds, filespace;
+    hid_t        dset_id = H5I_INVALID_HID, filespace = H5I_INVALID_HID;
     hsize_t      size[RANK];
     base_t       base, last;
-    unsigned int ncols, last_step, step;
-
-    assert(which < s->ndatasets);
+    unsigned int ncols, step;
+    int          i;
 
-    ds = s->dataset[which];
+    if (which >= s->ndatasets) {
+        fprintf(stderr, "the dataset order is bigger than the number of datasets");
+        TEST_ERROR;
+    }
 
-    if (H5Drefresh(ds) < 0)
-        errx(EXIT_FAILURE, "H5Drefresh failed");
+    dset_id = s->dataset[which];
 
-    filespace = H5Dget_space(ds);
+    /* Attempt to check the availablity of the chunks for a number time before reporting it as a failure */
+    for (i = 0; i < NUM_ATTEMPTS; i++) {
+        if (H5Drefresh(dset_id) < 0) {
+            fprintf(stderr, "H5Drefresh failed\n");
+            TEST_ERROR;
+        }
 
-    if (filespace == badhid)
-        errx(EXIT_FAILURE, "H5Dget_space failed");
+        if ((filespace = H5Dget_space(dset_id)) < 0) {
+            fprintf(stderr, "H5Dget_space failed\n");
+            TEST_ERROR;
+        }
 
-    if (H5Sget_simple_extent_dims(filespace, size, NULL) < 0)
-        errx(EXIT_FAILURE, "H5Sget_simple_extent_dims failed");
+        if (H5Sget_simple_extent_dims(filespace, size, NULL) < 0) {
+            fprintf(stderr, "H5Sget_simple_extent_dims failed\n");
+            TEST_ERROR;
+        }
 
-    ncols = (unsigned)(size[1] / s->chunk_dims[1]);
-    if (ncols < hang_back)
-        goto out;
+        ncols = (unsigned)(size[1] / s->chunk_dims[1]);
 
-    last_step = ncols - hang_back;
+        /* Make sure the chunks show up on the reader side.  Otherwise sleep a while and try again */
+        if (ncols >= last_step)
+            break;
+        else
+            decisleep(1);
+    }
 
-    for (step = *stepp; step <= last_step; step++) {
-        const unsigned ofs = step % 2;
+    if (i == NUM_ATTEMPTS) {
+        fprintf(stderr, "chunk verification reached the maximal number of attempts");
+        TEST_ERROR;
+    }
 
+    for (step = finished_step; step < last_step; step++) {
         dbgf(1, "%s: which %u step %u\n", __func__, which, step);
 
         if (s->two_dee) {
             size[0]  = s->chunk_dims[0] * (1 + step);
             size[1]  = s->chunk_dims[1] * (1 + step);
-            last.row = s->chunk_dims[0] * step + ofs;
```

Comment:
In the original test, we documented this in the test document:
•	For the reader: for some datasets the program explicitly chooses not to set the offset of a hyperslab at a chunk boundary. However, still each time the values of one full chunk size are read and verified. So effectively the data values across the chunks are verified.
Now we use the named pipe to communicate between the reader and the writer, this is no longer valid. A question may be to Vailin and John, should we update the test document some time to reflect this?

Reply:
Yes, the "tests" inventory document needs to be updated.  This is true not just for the bigset test but also other tests modified due to named pipes and also new testing operations added.

Reply:
I put it back, but added a command-line option to enable it in case that it might affect the performance measurement.